api/src/reports/report.schema.ts

```diff
@@ -0,0 +1,19 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { HydratedDocument, Schema as MongooseSchema } from 'mongoose';
+
+export type ReportDocument = HydratedDocument<Report>;
+
+@Schema({ timestamps: true })
+export class Report {
+  id: string;
+
+  @Prop({ type: MongooseSchema.Types.ObjectId, ref: 'User', required: true })
+  creatorId: string;
+
+  @Prop({ type: MongooseSchema.Types.ObjectId, ref: 'User', required: true })
+  targetId: string;
+
+  createdAt: Date;
```

Comment:
You should add `updatedAt` as well. I assume you didn't because there's no way to update a report, which is fair. But adding it here isn't actually what causes it to be populated in the db. That happens because of the `timestamps: true`. Since it's already being added to the db, we might as well have the class match what's in the db.